src/sentry/integrations/slack/notifications.py

```diff
@@ -116,31 +117,17 @@ def get_channel_and_integration_by_team(
     return {external_actor.external_id: external_actor.integration}
 
 
-def get_channel_and_token_by_recipient(
+def get_integrations_by_channel_by_recipient(
     organization: Organization, recipients: Iterable[Team | User]
-) -> Mapping[Team | User, Mapping[str, str]]:
-    output: MutableMapping[Team | User, MutableMapping[str, str]] = defaultdict(dict)
+) -> MutableMapping[Team | User, Mapping[str, Integration]]:
+    output: MutableMapping[Team | User, Mapping[str, Integration]] = defaultdict(dict)
     for recipient in recipients:
         channels_to_integrations = (
             get_channel_and_integration_by_user(recipient, organization)
             if isinstance(recipient, User)
             else get_channel_and_integration_by_team(recipient, organization)
         )
-        for channel, integration in channels_to_integrations.items():
-            try:
-                token = integration.metadata["access_token"]
-            except AttributeError as e:
```

Comment:
So if `access_token` doesn't exist, this actually results in a `KeyError` so this try/except was not doing anything. As a result, I am removing it. If for some reason `access_token` does not exist, we should be letting it blow up and sending the error to Sentry.